homework6/src/main/java/Main.java

```diff
@@ -0,0 +1,300 @@
+        System.out.println("\nElements of the set: ");
+        randomStringsSet.forEach(System.out::println);
+        System.out.println("\nShortest String: " + shortestString(randomStringsSet));
+
+    }
+
+    /**
+     * this method finds the shortest String in the Set
+     *
+     * @param set Set of String
+     * @return String type
+     */
+    private static String shortestString(Set<String> set) {
+        return set.stream().min(Comparator.comparingInt(String::length)).get();
+    }
+
+    /**
+     * color is an Optional characteristic at Car objects
+     * It is not necessary to define
+     * It can be an empty field
+     *
+     * @param carList List of Car
+     * @return List of String
+     */
+    private static List<String> colorIsPresent(List<Car> carList) {
+        return carList
+                .stream()
+                .map((s) -> s.getColor())
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * this method verifies that there is a car in List with a power of greater than 250
+     *
+     * @param carList List of Car
+     * @return boolean value
+     */
+    private static boolean checkIf(List<Car> carList) {
+        return carList
+                .stream()
+                .map(s -> s.getPower())
+                .anyMatch(s -> s > 250);
+    }
+
+    /**
+     * this method sorts the List in reverse order
+     *
+     * @param set Set of String
+     * @return List of String
+     */
+    private static List<String> sortingStringsInReverseOrder(Set<String> set) {
+        return set
+                .stream()
+                .sorted(Comparator.comparing(String::valueOf).reversed())
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * this method sorts the List by the "make"
+     *
+     * @param set Set of Car
+     * @return List of Car
+     */
+    private static List<Car> sortingCustomObjects(Set<Car> set) {
+        return set
+                .stream()
+                .sorted(Comparator.comparing(Car::getMake).reversed())
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * this method creates a Map with random numbers, and define the number of elements, which have value over 10
+     *
+     * @param n the number of the Integers
+     */
+    private static void nElementInMap(int n) {
+        Random random = new Random();
+        int index = 1;
+        Map<Integer, Integer> map = new HashMap<>();
+
+        while (index <= n) {
+            map.put(index, random.nextInt());
+            index++;
+        }
+
+        System.out.println("\nThe content of the map: " + map);
+
+        long nr = map
+                .entrySet()
+                .stream()
+                .filter(i -> i.getValue() > 10)
+                .count();
+
+        System.out.println("Number of elements have value over 10: " + nr);
```

Comment:
please try integrating the LOGGER for next time since Syso is rarely used in applications: https://www.vogella.com/tutorials/Logging/article.html